src/main/resources/site/parts/button/button.ts

```diff
@@ -0,0 +1,84 @@
+import {Request, Response} from "enonic-types/controller";
+import {Content} from "enonic-types/content";
+import {Button} from "../../content-types/button/button";
+import {Component} from "enonic-types/portal";
+import {ButtonPartConfig} from "./button-part-config";
+
+const {
+  attachmentUrl,
+  getComponent,
+  pageUrl
+} = __non_webpack_require__('/lib/xp/portal')
+const {
+  render
+} = __non_webpack_require__('/lib/thymeleaf')
+const {
+  renderError
+} = __non_webpack_require__('/lib/ssb/error/error')
+
+const content = __non_webpack_require__('/lib/xp/content')
+const util = __non_webpack_require__('/lib/util')
+const view = resolve('./button.html')
+
+exports.get = function(req: Request): Response {
+  try {
+    const part: Component<ButtonPartConfig> = getComponent()
+    const buttonsIds: Array<string> = part.config.button ? util.data.forceArray(part.config.button) : []
+    return renderPart(req, buttonsIds)
+  } catch (e) {
+    return renderError(req, 'Error in part', e)
+  }
+}
+
+exports.preview = (req: Request, id: string) => renderPart(req, [id])
+
+function renderPart(req: Request, buttonIds: Array<string>): Response {
+  let buttons: Array<ButtonShape> = []
+
+  buttonIds.map((key: string) => {
+    const button: Content<Button> | null = content.get({
+      key
+    })
+
+    if (button && button.data.link) {
+      const target: Content<any> | null = content.get({
```

Comment:
Do we need <any> ? Might work with just Content (defined in getHref func as well)

Reply:
Correct, that's better as long as we don't need any fields specific to a content type.